packages/knip/src/plugins/vitest/index.ts

```diff
@@ -84,6 +111,30 @@ export const resolveConfig: ResolveConfig<ViteConfigOrFn | VitestWorkspaceConfig
 
   const configs = await getConfigs(localConfig);
 
+  // Resolve external project files from glob patterns in vitest projects configuration
+  const externalProjectFiles: string[] = [];
+  for (const cfg of configs) {
+    if (cfg.test?.projects) {
+      for (const project of cfg.test.projects) {
+        if (typeof project === 'string') {
+          // Expand glob patterns to actual configuration file paths
+          const projectFiles = await _glob({
+            cwd: options.cwd,
+            patterns: [project],
+            gitignore: false,
+          });
+          externalProjectFiles.push(...projectFiles);
+        }
+      }
+    }
+  }
+
+  // Process external project configuration files and assign dependencies to correct workspace
+  for (const projectFile of externalProjectFiles) {
+    inputs.add(toEntry(projectFile));
+    inputs.add(toConfig('vitest', projectFile));
+  }
```

Comment:
Looks like we can skip creating `externalProjectFiles` and add this to inputs directly above:

```ts
inputs.add(toConfig('vitest', projectFile, { containingFilePath: options.configFilePath }));
```

These items are added as entry files automatically (i.e. `toEntry` is redundant).